homeassistant/components/energy/sensor.py

```diff
@@ -271,8 +272,10 @@ def _update_cost(self) -> None:
 
         elif self._adapter.source_type == "gas":
             valid_units = VALID_ENERGY_UNITS_GAS
-            # No conversion for gas.
-            default_price_unit = None
+            if self.hass.config.units is METRIC_SYSTEM:
+                default_price_unit = UnitOfVolume.CUBIC_METERS
+            else:
+                default_price_unit = UnitOfVolume.CENTUM_CUBIC_FEET
```

Comment:
I would like to have a second opinion on this, since it will now do a conversion in cases where it didn't before.

Reply:
The problem is that we allow gas to be measured either in a volume or as energy, so I don't think this will work (that's what the comment is trying to say).